src/account/operations/transfer/high_level/burn_native_token.rs

```diff
@@ -0,0 +1,204 @@
+        account
+            .unspent_outputs()
+            .values()
+            .into_iter()
+            .filter(|output_data| match &output_data.output {
+                Output::Alias(output) => output.alias_id().or_from_output_id(output_data.output_id) == alias_id,
+                Output::Foundry(output) => output.id() == foundry_id,
+                _ => false,
+            })
+            .for_each(|output_data| match &output_data.output {
+                Output::Alias(_) => existing_alias_output_data = Some(output_data),
+                Output::Foundry(output) => existing_foundry_output = Some(output),
+                _ => unreachable!("We checked if it's an alias or foundry output before"),
+            });
+
+        let existing_alias_output_data = existing_alias_output_data
+            .ok_or_else(|| Error::BurningFailed("Required alias output for token id not found".to_string()))?
+            .clone();
+
+        let existing_foundry_output = existing_foundry_output
+            .ok_or_else(|| Error::BurningFailed("Required foundry output for token id not found".to_string()))?
+            .clone();
+
+        drop(account);
+
+        if let Output::Alias(alias_output) = &existing_alias_output_data.output {
+            // Create the new alias output with the same feature blocks, just updated state_index
+            let mut new_alias_output_builder = AliasOutputBuilder::new(existing_alias_output_data.amount, alias_id)?
+                .with_state_index(alias_output.state_index() + 1)
+                .with_foundry_counter(alias_output.foundry_counter());
+
+            for unlock_condition in alias_output.unlock_conditions().clone().into_iter() {
+                new_alias_output_builder = new_alias_output_builder.add_unlock_condition(unlock_condition);
+            }
+
+            for feature_block in alias_output.feature_blocks().iter() {
+                new_alias_output_builder = new_alias_output_builder.add_feature_block(feature_block.clone());
+            }
+            for immutable_feature_block in alias_output.immutable_feature_blocks().iter() {
+                new_alias_output_builder =
+                    new_alias_output_builder.add_immutable_feature_block(immutable_feature_block.clone());
+            }
+
+            let TokenScheme::Simple(foundry_simple_ts) = existing_foundry_output.token_scheme();
+            let outputs = vec![
+                Output::Alias(new_alias_output_builder.finish()?),
+                Output::Foundry(
+                    FoundryOutputBuilder::new(
+                        minimum_storage_deposit_foundry(&byte_cost_config)?,
+                        foundry_id.serial_number(),
+                        token_id.token_tag(),
+                        TokenScheme::Simple(SimpleTokenScheme::new(
+                            foundry_simple_ts.circulating_supply(),
+                            foundry_simple_ts.melted_tokens() + burn_token_amount,
```

Comment:
Before we just add it here, there should be a check that we even have this amount of native tokens available in the inputs, because if we would only have for example `10` native tokens but want to burn `20`, it will not be valid

Reply:
Right, but since I'm not passing in inputs explicitly, does this mean `try_select_inputs` cannot detect insufficient amount for native tokens?

Reply:
Ah yeah `try_select_inputs` should return an error then